File: JAVA/src/client/CalculatorClient.java
package client;

import soapclient.Calculator;
import soapclient.CalculatorSoap;

public class CalculatorClient {
    public static void main(String[] args) {
        // Crear una instancia del servicio
        Calculator service = new Calculator();
        CalculatorSoap port = service.getCalculatorSoap();

        // Invocar operaciones del servicio
        int a = 10;
        int b = 5;

        System.out.println("Operaciones con SOAP Calculator:");
        System.out.println("Suma (" + a + " + " + b + "): " + port.add(a, b));
        System.out.println("Resta (" + a + " - " + b + "): " + port.subtract(a, b));
        System.out.println("Multiplicación (" + a + " * " + b + "): " + port.multiply(a, b));
        System.out.println("División (" + a + " / " + b + "): " + port.divide(a, b));
    }
}

File: PYTHON/calculator_client.py
from zeep import Client

def main():
    # URL del WSDL
    wsdl_url = "http://www.dneonline.com/calculator.asmx?WSDL"

    # Crear cliente SOAP
    client = Client(wsdl=wsdl_url)

    # Llamar al método Add
    result_add = client.service.Add(15, 10)
    print(f"Resultado de la suma: {result_add}")

    # Llamar al método Subtract
    result_subtract = client.service.Subtract(20, 5)
    print(f"Resultado de la resta: {result_subtract}")

    # Llamar al método Multiply
    result_multiply = client.service.Multiply(6, 7)
    print(f"Resultado de la multiplicación: {result_multiply}")

    # Llamar al método Divide
    result_divide = client.service.Divide(50, 5)
    print(f"Resultado de la división: {result_divide}")

if __name__ == "__main__":
    main()
